### PR TITLE
Generation of CHANGELOG filtering merges by base branch

### LINKED ISSUE #42: Filter master merges when generating the CHANGELOG
This changelog generator works by finding PR merges and adding them to the latest release. This works fine when we assume all merges are done to `master`, but is not enough when there are merges to other branches.

### Example

Alice is working on a new authentication feature in its own feature branch. It is not yet ready to be merged to master. Bob notices a bug and creates a PR to that feature branch with a fix. Alice merges the PR and continues working on the feature.

Here's the current git tree:

```
*   432e63f - (HEAD -> feature/new-auth) Merge branch 'bugfix/fix-auth-bug' into feature/new-auth <Alice>
|\
| * 5d9ecbb - (bugfix/fix-auth-bug) Fix auth bug <Bob>
* | 0fad650 - Improve auth test coverage <Alice>
|/
* 3669716 - Implement new auth approach <Alice>
* c04d5c3 - (master) Fix typo <Charlie>
```

Now, Charlie creates a new release and runs `github-changelog-generator` to update the changelog with the new release. Here's the current git tree:

```
*   12f3f12 - (master) Release 1.23 <Charlie>
| *   432e63f - (feature/new-auth) Merge branch 'bugfix/fix-auth-bug' into feature/new-auth <Alice>
| |\
| | * 5d9ecbb - (bugfix/fix-auth-bug) Fix auth bug <Bob>
| * | 0fad650 - Improve auth test coverage <Alice>
| |/
| * 3669716 - Implement new auth approach <Alice>
| |
|/
* c04d5c3 - Fix typo <Charlie>
```

Here's the generated CHANGELOG.md:

```
# Changelog

## [1.23](releases/tag/v1.23) (2017-10-17)
- Fix auth bug [#45] (bob)
- Fix typo [#41] (charlie)
```

The generated changelog is wrong, because `Fix auth bug` is not yet merged to `master` and therefore is not actually part of release 1.23. The correct changelog for this release would have been:

```
# Changelog

## [1.23](releases/tag/v1.23) (2017-10-17)
- Fix typo [#41] (charlie)
```

---

It would be interesting to allow filtering not only for merges to `master`, but also for merges to another designated branch. So, if a team runs releases from branch `nightly` or `dev` instead of `master`, they could still use this generator.

### PR DESCRIPTION
**Pull request by marianacapelo**
Closes #42.

New optional parameter to set base branch value - using 'master' as default.
Updated getPullRequestsPage function to use parameter value in pullRequests.getAll call.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ program
   .option('-r, --repo <name>', '[required] name of the repository')
   .option('-f, --future-release <version>', '[optional] specify the next release version')
   .option('-t, --future-release-tag <name>', '[optional] specify the next release tag name if it is different from the release version')
+  .option('-b, --branch <name>', '[optional] specify the base (target) branch name - by default, uses pull requests into master')
```
Comment (rplopes) [inline src/index.js]: Maybe `--base-branch` sends a clearer message about the purpose of this option? WDYT?

Also, for brevity's sake, "by default, uses pull requests into master" could be simplified to something like "master by default".

File: src/index.js [hunk 2]
```diff
@@ -28,6 +29,7 @@ program
 const concurrency = 20;
 const { futureRelease, owner, repo } = program;
 const futureReleaseTag = program.futureReleaseTag || futureRelease;
+const branch = program.branch || 'master';
```
Comment (rplopes) [inline src/index.js]: We can call it `base`, to be consistent with GitHub's naming.

**Comment by rplopes** [comment]
Also closes #24.

File: src/index.js (hunk 1 above)
Comment (marianacapelo) [reply src/index.js]: 👍 I agree

**Comment by marianacapelo** [comment]
@rplopes, I've just updated the PR. Thank you for your feedback

File: src/index.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ program
   .option('-r, --repo <name>', '[required] name of the repository')
   .option('-f, --future-release <version>', '[optional] specify the next release version')
   .option('-t, --future-release-tag <name>', '[optional] specify the next release tag name if it is different from the release version')
+  .option('-b, --base-branch <name>', '[optional] specify the base branch name - master by default')
```
Comment (rplopes) [inline src/index.js]: The program options should be sorted alphabetically within each required/optional group, for easier reference.

File: src/index.js [hunk 4]
```diff
@@ -28,6 +29,7 @@ program
 const concurrency = 20;
 const { futureRelease, owner, repo } = program;
 const futureReleaseTag = program.futureReleaseTag || futureRelease;
+const base = program.baseBranch || 'master';
```
Comment (rplopes) [inline src/index.js]: One last small detail: we prefer to sort variable declarations alphabetically whenever possible, for better maintainability.

**Comment by marianacapelo** [comment]
@rplopes again, thank you. I updated the order (of new option and variable declaration) as you said, and it does look better 👍

**Pull request merged**